module/zfs/dsl_prop.c

```diff
@@ -123,7 +123,7 @@ dsl_prop_get_dd(dsl_dir_t *dd, const char *propname,
 		/* Check for a iuv value. */
 		err = zap_lookup(mos, dsl_dir_phys(dd)->dd_props_zapobj,
 		    iuvstr, intsz, numints, buf);
-		if (dsl_prop_known_index(zfs_name_to_prop(propname),
+		if (err == 0 && dsl_prop_known_index(zfs_name_to_prop(propname),
```

Comment:
I just noticed that there is a small optimization opportunity here. We can replace `zfs_name_to_prop(propname)` with `prop` so that this function is not being called repeatedly.

Reply:
I pushed a second commit for this.